datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -136,6 +136,54 @@ def validate_a_node(
     )
 
 
+@router.post("/nodes/{name}/validate/", response_model=NodeValidation)
+def revalidate_a_node(
+    name: str,
+    session: Session = Depends(get_session),
+) -> NodeValidation:
+    """
+    Revalidate a single existing node and update its status appropriately
+    """
+    node = get_node_by_name(session, name)
+    current_node_revision = node.current
+    if current_node_revision.type == NodeType.SOURCE:
+        current_node_revision.status = NodeStatus.VALID
```

Comment:
This is dangerous, if the source node does not match the underlying table anymore. Right?

Reply:
You're right, I hadn't considered that reflection could be setting a source node as invalid. Good catch!

Reply:
@agorajek thinking about this a bit more, since we're going to leave this as an admin endpoint, is it valuable to leave this here as an escape hatch for when a reflection service is acting up?

Reply:
Yes agree.